### PR TITLE
feat(presentation): Add removable and downloadable attributes to payload.

### LINKED ISSUE #13238: Add a new setting to block removal of presentations
It would be good to have a setting to disable the delete presentation feature.

This is useful when we want to ensure that presentations will be available for the entire meeting ( in case of pre-uploaded presentations it's more important as moderator don't have the file to re-upload ).

I propose that we add a flag similar to `presentation.allowDownloadable` but with name `presentation.allowRemove`.

When this flag is true, we could hide the delete presentation button:
image

### PR DESCRIPTION
**Pull request by GuiLeme**
### What does this PR do?

It has been added 2 new attributes to create room endpoint's payload: `removable`, and `downloadable`. The first one is related to not being able to remove the preloaded presentation, it serves as a protection as the moderator could delete it unintentionally. As for the downloadable property, it defines if the attendees will be authorized to download the presentation (this attribute set the initial value, as it can be customized by moderator in presentation window).
So the following request:

image

Should result as:

image

### Closes Issue(s)

Closes #13238

### CODE REVIEW

**Comment by antobinary** [comment]
Hi @GuiLeme !
Thank you for the PR! The code looks good to me when I read through it. I am yet to pull and test.
I changed the destination branch to `v2.5.x-release` as it is more suitable for merging these enhancements!

I am not spotting if we need to update any docs in https://docs.bigbluebutton.org . Did you resort to any docs when developing these changes?

**Comment by GuiLeme** [comment]
Hey,  @antobinary !
For this issue, I didn't use any docs. I don't think we need to update anything in the docs for now.

File: bbb-common-message/src/main/scala/org/bigbluebutton/common2/msgs/PresentationPodsMsgs.scala [hunk 1]
```diff
@@ -228,6 +230,7 @@ case class PresentationConversionRequestReceivedEventMsgBody(
     current:        Boolean,
     presName:       String,
     downloadable:   Boolean,
+    removable:      Boolean,
```
Comment (antobinary) [inline bbb-common-message/src/main/scala/org/bigbluebutton/common2/msgs/PresentationPodsMsgs.scala:233]: These events are also used by bbb-html5, see https://github.com/bigbluebutton/bigbluebutton/blob/v2.5.x-release/bigbluebutton-html5/imports/api/presentations/server/eventHandlers.js 

I get

```
I20220203-03:03:59.714(0)? info: Updated presentation conversion status=SUPPORTED_DOCUMENT id=d2d9a672040fbde2a47a10bf6c37b6a4b5ae187f-1643857434602 meeting=1c6637a8f2e1f75e06ff9984894d6bd16a3a36a9-1643857434601
I20220203-03:04:00.408(0)? error: PresentationConversionCompletedEvtMsg: Error: Match error: Unknown key in field removable
I20220203-03:04:00.409(0)?     at check (packages/check/match.js:36:17)
I20220203-03:04:00.409(0)?     at addPresentation (imports/api/presentations/server/modifiers/addPresentation.js:32:3)
I20220203-03:04:00.409(0)?     at EventEmitter.handlePresentationAdded (imports/api/presentations/server/handlers/presentationAdded.js:13:10)
I20220203-03:04:00.409(0)?     at EventEmitter.emitAsync (/home/bigbluebutton/src/bigbluebutton-html5/node_modules/eventemitter2/lib/eventemitter2.js:463:31)
I20220203-03:04:00.409(0)?     at MeetingMessageQueue.handleTask (imports/startup/server/redis.js:105:10)
I20220203-03:04:00.409(0)?     at imports/startup/server/redis.js:117:28
I20220203-03:04:00.409(0)?     at Queue.start (/home/bigbluebutton/src/bigbluebutton-html5/node_modules/queue/index.js:145:17)
I20220203-03:04:00.410(0)?     at Queue.<computed> [as push] (/home/bigbluebutton/src/bigbluebutton-html5/node_modules/queue/index.js:59:12)
I20220203-03:04:00.410(0)?     at MeetingMessageQueue.add (imports/startup/server/redis.js:116:16)
I20220203-03:04:00.410(0)?     at RedisPubSub.handleMessage (imports/startup/server/redis.js:318:63)
I20220203-03:04:00.410(0)?     at runWithEnvironment (packages/meteor.js:1320:24)
```

not matching object structure. Please commit the files you had to modify in bbb-html5 too

Comment (GuiLeme) [reply bbb-common-message/src/main/scala/org/bigbluebutton/common2/msgs/PresentationPodsMsgs.scala:233]: Hey @antobinary , sorry about it, that passed without me noticing it, I've just made a commit with the other changes, apparently there is some conflict which I am going to tackle right now.

File: bigbluebutton-html5/imports/ui/components/presentation/presentation-uploader/component.jsx [hunk 2]
```diff
@@ -783,6 +783,8 @@ class PresentationUploader extends Component {
       ? cx(styles.itemAction, styles.itemActionRemove, styles.checked)
       : cx(styles.itemAction, styles.itemActionRemove);
 
+    const isRemovable = item.isRemovable
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/presentation/presentation-uploader/component.jsx:796]: -const isRemovable = item.isRemovable 

+const { isRemovable}  = item;

**Pull request merged**